Grouping aggregates

jacekkopecky opened the pull request:
This isn't quite complete as there is no UI for adding and manipulating grouping aggregates yet.

jacekkopecky committed:
server: accept metaanalysis.groupingColumn

jacekkopecky committed:
Metaanalysis: Introduce grouping table

Includes prelim selection and saving of a column, via a dropdown box
filled with the paper’s columns.

jacekkopecky committed:
metaanalysis: Populate and calculate grouping table

For now we just get a copy of every aggregate, and then calculate the
values by group

jacekkopecky committed:
formulas: use asterisk to indicated grouping aggregates

jacekkopecky committed:
metaanalysis: groupingAggregates saving and loading

jacekkopecky committed:
metaanalysis: display grouping aggregates better

jacekkopecky committed:
metaanalysis: allow grouping aggregates as parameters

Also disallow aggregates as parameters of graphs.

jacekkopecky committed:
metaanalysis: computing with grouping aggregates

Jnnybrrn committed:
Metaanalysis: Add/delete/move for groupedAggregates

Includes popupboxes for grouped aggregates:

- The aggregate now shows popupbox, similar to aggregates (including comments).
- The datum now shows a popupbox similar to computed column datum.

Jnnybrrn reviewed (dismissed):
Looks good to me - You need to check the last commit (add/delete/move gAggr's)

jacekkopecky committed:
:art:

includes whitespace changes which makes the patch dirtier than it would
otherwise be
